internal/pkg/pkg.go

```diff
@@ -145,14 +144,65 @@ func (p *Pkg) RelativePathTo(ancestorPkg *Pkg) (string, error) {
 // TODO: This does not support symlinks, so we need to figure out how
 // we should support that with kpt.
 func (p *Pkg) DirectSubpackages() ([]*Pkg, error) {
```

Comment:
A subpackage is a direct subpackage iff it exists in the top level folder of a package ? What if it exists within the directory ? We will be missing out such packages when do hydration using functions.

This equation must hold true.
Recursive sum of all direct subpackages = Total number of packages

Reply:
Yeah, this does find subpackages nested in directories. Direct here means it doesn't include subpackages nested inside other subpackages (so sub-subpackages). I have updated the description to make this clear.